spec/lib/tasks/data_hygiene/content_item_deduplicator_spec.rb

```diff
@@ -0,0 +1,44 @@
+require "rails_helper"
+require "tasks/data_hygiene/content_item_deduplicator"
+
+describe Tasks::DataHygiene::ContentItemDeduplicator do
+  let(:fake_stdout) { StringIO.new }
+
+  before do
+    @real_stdout = $stdout
+    $stdout = fake_stdout
+  end
+
+  after { $stdout = @real_stdout }
+
+  describe "#deduplicate" do
+    it "runs without issue" do
+      content_item = create(:content_item_with_content_id, locale: 'cy')
+      dupe = create(:content_item, content_id: content_item.content_id, locale: 'cy')
+      locale_variant = create(:content_item, content_id: content_item.content_id, locale: 'en')
+
+      expect { subject.deduplicate }.to change(ContentItem, :count).by(-1)
+
+      fake_stdout.rewind
+      output = fake_stdout.read
+      expect(output).to match(/2 duplicates found/)
```

Comment:
There are output matchers that let you assert on `stdout` and `stderr`. You could go fully end-to-end and also remove the need to overwrite them in the tests. Something like:

```ruby
expect { subject.deduplicate }.to output('2 duplicates found').to_stdout
# etc
```